### PR TITLE
Feat 4e - Stitched polls controller 🧵

### PR DESCRIPTION
**Pull request by zenajun**
Stitched together Chao's postPolls, Cher's insertDataToFB and Zena's postToSlack code. 😅💦

When a user submits a poll from the FE, we

receive the payload in the BE
Send the poll to Firebase
Send the poll to slack (with the Firebase ID attached).
Refactored some of the code to services.js

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -46,10 +46,8 @@ app.use("/channels", channelRouter);
 // // app.get("/", );
 app.use("/polls", pollsRouter);
 app.use("/result/:id", resultRouter);
-// postTestMessage(
-//   process.env.SLACKBOT_TEST_CHANNEL,
-//   "Changed this to postTestMessage 123"
-// );
+
+app.use("/polls", pollsRouter);
```
Comment (MustafaHaddara) [inline src/index.js]: is this duplicated? what's the difference between this and line 47?

File: src/routes/polls/polls.controller.js [hunk 2]
```diff
@@ -22,11 +8,7 @@ const postPolls = function(req, res) {
   res.send(`Received new Poll: ${body.message}`);
 
   // Function call to insert data
-  insertData("SLACKBOT_TEST", body)
-    .then(() => {
-      console.log(`Success`);
-    })
-    .catch(err => console.error(err));
+  insertDataToFB("SLACKBOT_TEST", body);
```
Comment (MustafaHaddara) [inline src/routes/polls/polls.controller.js]: at first i read this "facebook" and got confused haha 😅

File: src/routes/polls/polls.router.js [hunk 3]
```diff
@@ -1,5 +1,15 @@
 const express = require("express");
 
+const {
+  getPreviousPollsController,
+  createPollController
```
Comment (MustafaHaddara) [inline src/routes/polls/polls.router.js]: do these names actually match? I only see `postPolls` and `getPolls` in the exports for `polls.controller`...

File: src/index.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ const { errorHandler } = require("./middleware/error-handler");
 const { channelRouter } = require("./routes/channels/channels.router");
 const { pollsRouter } = require("./routes/polls/polls.router");
 const { resultRouter } = require("./routes/result/result.router");
+const { pollsRouter } = require("./routes/polls/polls.router");
```
Comment (MustafaHaddara) [inline src/index.js]: this looks like the same as line 17?

File: src/index.js (hunk 1 above)
Comment (zenajun) [reply src/index.js]: Oopsie. Fixed.

File: src/index.js (hunk 4 above)
Comment (zenajun) [reply src/index.js]: I think I fixed everything now. Thanks for catching

File: src/routes/polls/services.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    console.log(`Error posting message: `, err);
+  }
+};
+
+// Function to insert question to Firebase
+const insertDataToFB = async (col, data) => {
+  try {
+    const doc = await db.collection(col).doc();
+    const id = doc.id;
+    doc[id] = id;
```
Comment (MustafaHaddara) [inline src/routes/polls/services.js]: why do you need this line? didn't you just read it from the same property name (`id`) on the line above?

**Pull request merged**